pkg/apis/monitoring/v1/types.go

```diff
@@ -426,6 +432,36 @@ type PrometheusRuleExcludeConfig struct {
 	RuleName string `json:"ruleName"`
 }
 
+// PrometheusObjectReference references a PodMonitor, ServiceMonitor, Probe or PrometheusRule object.
+type PrometheusObjectReference struct {
+	// API version of the referent.
+	APIVersion string `json:"apiVersion"`
+	// Kind of the referent.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	// +kubebuilder:validation:Enum=PrometheusRule;ServiceMonitor;PodMonitor;Probe
+	Kind string `json:"kind"`
+	// Namespace of the referent.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/
+	Namespace string `json:"namespace"`
+	// Selector for PodMonitor, ServiceMonitor, Probe or PrometheusRule objects based on name OR labels.
+	// matchLabels and matchExpressions are ANDed together and the result is ORed with matchNames.
+	Selector NameOrLabelsSelector `json:"selector"`
+}
```

Comment:
Reading https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#multiple-resource-reference, the following approach would be more recommended:

```suggestion
type ObjectReference struct {
	// Group of the referent.
	Group string `json:"group"`
	// Resource of the referent.
	// +kubebuilder:validation:Enum=prometheusrules;servicemonitors;podmonitors;probes
	Resource string `json:"resource"`
	// Namespace of the referent.
	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/
	Namespace string `json:"namespace"`
}
```